packages/cli/src/commands/new.ts

```diff
@@ -99,26 +92,34 @@ export class New extends Command {
     debug("args: ", args)
     debug("flags: ", flags)
 
-    await this.determinePkgManagerToInstallDeps(flags)
     const {pkgManager, shouldInstallDeps} = this
-
     const shouldUpgrade = !flags["skip-upgrade"]
+
     if (shouldUpgrade) {
+      const spinner = log
+        .spinner(log.withBrand("Checking if new Blitz release is available"))
+        .start()
       const wasUpgraded = await this.maybeUpgradeGloballyInstalledBlitz()
+      spinner.succeed()
       if (wasUpgraded) {
+        log.success("Blitz was sucessfully upgraded to latest version")
         this.rerunButSkipUpgrade(flags, args)
         return
       }
+
+      log.success("You already have the latest version")
     }
 
+    await this.determinePkgManagerToInstallDeps(flags)
+
     try {
       const destinationRoot = require("path").resolve(args.name)
       const appName = require("path").basename(destinationRoot)
 
       const form = await this.determineFormLib(flags)
 
       const {"dry-run": dryRun, "no-git": skipGit} = flags
-      const needsInstall = dryRun || !shouldInstallDeps
+      const needsInstall = !dryRun && shouldInstallDeps
```

Comment:
This condition was wrong isn't it ? 
We should install if we are not dry running and if `shouldInstallDeps` is truthy

Reply:
Here's an issue related to `dry run`: https://github.com/blitz-js/blitz/issues/2118. I think that are more problems with dry run flag, but yeah, looks like the condition was wrong